sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/models/BinaryData.java

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.amqp.models;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+/**
+ * Binary representation of amqp message body.
+ */
+public final class BinaryData {
+    private final byte[] data;
+    /**
+     *Create {@link BinaryData} instance with given byte array data.
+     *
+     * @param data to use.
+     */
+    public BinaryData(byte[] data) {
```

Comment:
Two questions for this class.
1. Why do we want to wrap the data of a byte array into a class if we don't provide extra functionality than a byte array?
2. The constructor and every call of getData copy the bytes. This might waste some memory.

Reply:
1. We will add serializer support but we need design/discussion around it, next month we will add more functionality in it. This will mimic how dotnet jave https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/core/Azure.Core.Experimental#binary-data
And for API completeness , we want to this type which will support similar feature like dotnet type.

2. We have a check which does not allow to return internal `byte[]` representation because it is mutable.